Dockerfile.cpu

```diff
@@ -0,0 +1,62 @@
+FROM ubuntu:18.04 as build-stage
+
+WORKDIR /usr/src/app
+
+ENV LANG="C.UTF-8" LC_ALL="C.UTF-8" PATH="/opt/venv/bin:$PATH" PIP_NO_CACHE_DIR="false" CFLAGS="-mavx2 -mf16c" CXXFLAGS="-mavx2 -mf16c"
+
+RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
+    python3 python3-pip python3-venv python3-dev wget make g++ libblas-dev liblapack-dev patch \
+    build-essential zlib1g-dev libsdl2-dev libjpeg-dev nasm tar libbz2-dev libgtk2.0-dev cmake \
+    git libfluidsynth-dev libgme-dev libopenal-dev timidity libwildmidi-dev unzip libboost-all-dev \
+    libcanberra-gtk-module libopenmpi-dev libosmesa6-dev libgl1-mesa-glx libglfw3 && \
+    rm -rf /var/lib/apt/lists/*
+
+RUN export DISPLAY=:0
+
+COPY requirements.txt .
+
+RUN python3 -m venv /opt/venv && \
+    python3 -m pip install pip==19.3.1 pip-tools==4.2.0 wheel==0.33.6 setuptools==44.0.0
+
+RUN python3 -m piptools sync
+
+COPY . .
+
+FROM ubuntu:18.04 as dev-stage
+
+WORKDIR /usr/src/app
+
+ENV LANG="C.UTF-8" LC_ALL="C.UTF-8" PATH="/opt/venv/bin:$PATH" PIP_NO_CACHE_DIR="false"
+
+RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
+    python3 python3-pip python3-venv python3-dev wget make g++ libblas-dev liblapack-dev patch \
+    build-essential zlib1g-dev libsdl2-dev libjpeg-dev nasm tar libbz2-dev libgtk2.0-dev cmake \
+    git libfluidsynth-dev libgme-dev libopenal-dev timidity libwildmidi-dev unzip libboost-all-dev \
+    libcanberra-gtk-module libopenmpi-dev libosmesa6-dev libgl1-mesa-glx libglfw3 && \
+    rm -rf /var/lib/apt/lists/*
+
+RUN export DISPLAY=:0
+
+COPY requirements.txt .
+
+RUN python3 -m venv /opt/venv && \
+    python3 -m pip install pip==19.3.1 pip-tools==4.2.0 setuptools==44.0.0
+
+RUN echo "https://download.pytorch.org/whl/cpu/torch-1.3.0%2Bcpu-cp36-cp36m-linux_x86_64.whl        \
+          --hash=sha256:ce648bb0c6b86dd99a8b5598ae6362a066cca8de69ad089cd206ace3bdec0a5f            \
+          \n                                                                                        \
+          https://download.pytorch.org/whl/cpu/torchvision-0.4.1%2Bcpu-cp36-cp36m-linux_x86_64.whl  \
+          --hash=sha256:593ad12c3c8ce16068566c9eb2bfb39f4834c89a2cc9f0b181e9121b06046b3e            \
+          \n" >> requirements.txt
+
+RUN python3 -m piptools sync
+
+RUN git clone https://github.com/openai/baselines.git && \
+    cd baselines && \
+    python3 -m pip install -e . && \
+    rm -rf baselines
```

Comment:
Capturing from chat:
- get tarball snapshot
- verify checksum

and always (!) pin deps to specific versions, gitshas, and related.

Reply:
Changed